lang/java/avro/src/main/java/org/apache/avro/generic/GenericData.java

```diff
@@ -590,6 +590,10 @@ protected void toString(Object datum, StringBuilder buffer, IdentityHashMap<Obje
       seenObjects.put(datum, datum);
       toString(datum, buffer, seenObjects);
       seenObjects.remove(datum);
+    } else if (datum instanceof org.joda.time.DateTime){
```

Comment:
This definitely cannot be done this way.   With 1.9/master, joda-time must be optional and this re-introduces a direct dependency on a joda.time class.    I'll need to think about this more, but this cannot be merged as-is.

Reply:
Good point, I think we should get rid of Joda since it is officially deprecated.

Reply:
I agree with Fokko, maintaining both Joda and Jsr310 implementation in the code just unnecessarily increases the complexity. Is Joda library still maintained at all? I know that it will be a breaking change, but for 1.9 release it is fine I guess.

Reply:
Huge +1 let's get rid of Joda for once and for all!

Reply:
I'm so in favor of this! 👍